src/pages/costModels/costModel/updateMarkupDialog.tsx

```diff
@@ -62,76 +91,111 @@ class UpdateMarkupModelBase extends React.Component<Props, State> {
                 source_uuids: current.sources.map(provider => provider.uuid),
                 source_type: current.source_type === 'OpenShift Container Platform' ? 'OCP' : 'AWS',
                 markup: {
-                  value: this.state.markup,
+                  value: this.state.sign === '-' ? this.state.sign + this.state.markup : this.state.markup,
                   unit: 'percent',
                 },
               };
               updateCostModel(current.uuid, newState, 'updateMarkup');
             }}
             isDisabled={
               isNaN(Number(this.state.markup)) ||
-              Number(this.state.markup) === Number(current.markup.value || 0) ||
+              (Number(this.state.markup) ===
+                Number(
+                  current.markup.value.startsWith('-') ? current.markup.value.substring(1) : current.markup.value || 0
+                ) &&
+                this.state.sign === this.state.origSign) ||
```

Comment:
It appears we're parsing "+" or "-" strings in a lot of places. We could test if markup is negative, then apply the sign.